mcr_recovery_behaviors/ros/include/force_field_recovery/force_field_recovery.h

```diff
@@ -193,7 +193,8 @@ class ForceFieldRecovery : public nav_core::RecoveryBehavior
     bool is_oscillation_detection_initialized_;
 
     // A pointer to the transform listener sent by move_base
-    tf::TransformListener* tf_;
+    tf2_ros::Buffer* tf_buffer_;
+    tf2_ros::TransformListener* tf_listener_;
```

Comment:
it's not necessary to create a tf listener since it's not being used. It's sufficient to change `tf::TransformListener` to `tf2_ros::Buffer` here and in the header.
See [clear_costmap_recovery](https://github.com/ros-planning/navigation/commit/a2837b5a9dc6dd4b4da176fca7d899d6a3722bf8#diff-2805759b10da6a3d35bd136e6ed1e534)

Reply:
sorry, you can ignore my comment. As @DharminB said, it is required by `tf_buffer_` (which is used to transform the pointcloud). But the object needs to be deleted somewhere (probably in the destructor)